model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/schema_aware_transforms.proto

```diff
@@ -0,0 +1,73 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+// Protocol Buffers describing available Schema-Aware transforms.
+// These are transforms that can be configured using Beam Schema-compatible parameters.
+// Runners can override these transforms with a native implementation.
+
+syntax = "proto3";
+
+package org.apache.beam.model.pipeline.v1;
+
+option go_package = "github.com/apache/beam/sdks/v2/go/pkg/beam/model/pipeline_v1;pipeline_v1";
+option java_package = "org.apache.beam.model.pipeline.v1";
+option java_outer_classname = "SchemaAwareTransforms";
+
+import "org/apache/beam/model/pipeline/v1/beam_runner_api.proto";
+import "org/apache/beam/model/pipeline/v1/schema.proto";
+
+
+message StandardSchemaAwareTransforms {
+    // Payload for all of these: SchemaAwareTransformPayload
+    enum IOs {
+        ICEBERG_READ = 0 [(beam_urn) = "beam:schematransform:org.apache.beam:iceberg_read:v1"];
+        ICEBERG_WRITE = 1 [(beam_urn) = "beam:schematransform:org.apache.beam:iceberg_write:v1"];
+    }
+    enum Managed {
+        // Payload: ManagedSchemaTransformPayload
+        MANAGED = 0 [(beam_urn) = "beam:transform:managed:v1"];
+    }
+}
+
+
+// Payload for a Schema-aware PTransform.
+// This is a transform that is aware of its input and output PCollection schemas
+// and is configured using Beam Schema-compatible parameters.
+// The information available in the payload can be used by runners to override the schema-aware transform.
+message SchemaAwareTransformPayload {
+    // The schema of the configuration row used to override the transform
+    Schema expansion_schema = 1;
+    // The configuration used to override this transform.
+    // Must be compatible with the configuration schema, and decodable via beam:coder:row:v1.
+    bytes expansion_payload = 2;
+}
+
+// Payload for a ManagedSchemaTransform.
+// The information available in the payload can be used by runners to override the ManagedSchemaTransform.
+message ManagedSchemaTransformPayload {
+    // The underlying schema-aware transform's identifier.
+    string underlying_transform_identifier = 1;
+    // The managed transform's configuration Schema.
+    Schema expansion_schema = 2;
+    // The configuration used to override the managed transform.
+    // Must be compatible with the expansion schema, and decodable via beam:coder:row:v1.
+    bytes expansion_payload = 3;
+    // The underlying transform's configuration, represented as a YAML string.
+    string yaml_config = 4;
```

Comment:
Proto will make it easy to do upgrading since we can just pass that in. YAML will force us to build the Row when upgrading. I don't mind having both which will allow us to go either way.

Reply:
Keep in mind the encoded row here is the configuration for Managed transform (which includes the configuration for the underlying transform, in the form of a yaml string or URL).  

Replied the same to the comment below, but I believe the `yaml_config` field here is not for functional use. Just to more easily extract config values from the runner side.

Reply:
We'll be using `SchemaTransformPayload`, which means the yaml config will be found inside the encoded `configuration_row`